app/controllers/orders_controller.rb

```diff
@@ -17,17 +17,27 @@ def create
     end
   end
 
+  def show
+    render json: { 'order' => Order.find(params[:id]) }
+  end
+
   def update
+    return render json: { 'error' => 'You are not a driver' }, status: 422 unless @current_user.instance_of? Driver
     order = Order.find(params[:id])
-    order.driver_id = Driver.find(2).id # should be current driver
+    order.driver_id = @current_user.id
     order.state = 'in progress'
     order.save
-    client_email = Client.find(order.client_id).email
-    ClientMailer.welcome_email(params[:id], client_email).deliver_now if client_email
+    render json: { 'current order' => order.id }
```

Comment:
@domenic99 I think [it's better not to use spaces in JSON keys, as far as it complicates the process of parsing on the front end side by prohibiting usage of the dot notation](http://stackoverflow.com/questions/10311361/accessing-json-object-keys-having-spaces).
Also I suggest you to respond with the full JSON of the order.